server/zanata-frontend/src/editor/app/actions/phrases.js

```diff
@@ -128,12 +128,25 @@ function transUnitDetailToPhraseDetail (transUnitDetail, localeId) {
  * Get translations from a TransUnit in a consistent form (array of strings)
  *
  * This will always return an Array<String>, but the array may be empty.
+ *
+ * There is a problem with the API that means plurals for languages with a
+ * single plural form will use trans.content instead of trans.contents for the
+ * plural form. The result of this function has everything normalised to prevent
+ * it causing any problems.
  */
 function extractTranslations (source, trans) {
-  if (source.plural) {
-    return trans && trans.contents ? trans.contents.slice() : []
+  if (!trans) {
+    return []
+  }
+
+  // For non-plurals, just use the singluar content
+  // API uses trans.content if nplurals=1, so treat them like singular if
+  // trans.contents are empty.
+  if (!source.plural || isEmpty(trans.contents)) {
+    return trans.content ? [trans.content] : []
   }
-  return trans ? [trans.content] : []
+
+  return trans.contents ? trans.contents.slice() : []
```

Comment:
This code seems more complex than the Java equivalent in `TextContainer.getContents()`. I don't fully understand the code here (what is `source.plural`?), but perhaps this would be simpler:

    return trans.content ? [trans.content] : (trans.contents ? trans.contents.slice() : [])